### PR TITLE
feat: add MathWallet

### PR DESCRIPTION
**Pull request by mytangying**

### CODE REVIEW

**Comment by nickbytes** [comment]
Thanks for the contribution here @mytangying – we're currently discussing out strategy for adding new wallets to the repo and how we can best serve the RainbowKit community while also not being gatekeepers/maintainers of every wallet integration. I'll let you know when we have an update! Thanks for your patience!

**Comment by 0xzoz** [comment]
@nickbytes is there any update regarding the approach for adding new wallets?

I ask because im one of the Tally Ho community devs helping with integrations. As we ready for DAO launch we would love to have Tally Ho as an option for the awesome Rainbowkit library.

**Comment by Naxsun** [comment]
Getting added as a wallet connection option would be great. I'm currently in contact with two dapps that are planning to start using Rainbow kit, and would like to add Tally Ho as a specific wallet connection option.

**Comment by JVK-6935** [comment]
I came here to signal additional support for Tally Ho wallet.  I just tried connecting with it via Kwenta.io and Lyra on Optimism and it did not work there unlike Curve, Aave, Synthetix, etc. on Optimism that I also use.

**Comment by markdalgleish** [comment]
@mytangying Sorry for the delay on this, and thanks for the PR!

We've just re-worked RainbowKit internals in v0.7 so that wallets can be imported individually, ensuring our bundle size doesn't keep increasing as we continue to add more wallets.

Are you able to give us write access to your fork so that we can get you updated to the new API and apply some changes before merging?

**Comment by markdalgleish** [comment]
Also, could you update the logo in this PR so that it's a square rather than including the border radius? We apply a border radius ourselves in the UI and want to be sure it's consistent across all wallets.

**Comment by markdalgleish** [comment]
@0xzoz We're revisiting the wallet integrations at the moment if you're keen to open a PR.

**Comment by 0xzoz** [comment]
Cheers @markdalgleish. Let me know if you want me to run through some new docs or just do something similar to this

**Comment by markdalgleish** [comment]
@0xzoz Our format has changed a bit, but you can reference this PR which I've just migrated to the latest API and conventions: https://github.com/rainbow-me/rainbowkit/pull/550

**Comment by 0xzoz** [comment]
> @0xzoz Our format has changed a bit, but you can reference this PR which I've just migrated to the latest API and conventions: #550

@markdalgleish added #824 - took longer than expected due to trying to get Tally Ho as a connector in Wagmi

File: site/components/Provider/Provider.tsx
Comment (DanielSinclair) [inline site/components/Provider/Provider.tsx:33]: Please revert

File: packages/rainbowkit/src/wallets/walletConnectors/mathwallet/mathwallet.ts
Comment (DanielSinclair) [inline packages/rainbowkit/src/wallets/walletConnectors/mathwallet/mathwallet.ts:62]: Missing `qrCode` flag for the WalletConnect flow on Desktop

**Review by DanielSinclair — CHANGES_REQUESTED** [review]

**Comment by DanielSinclair** [comment]
Upon review, the WalletConnect flow on Desktop is not functioning as expected. Reference the review
Screenshot 2023-02-07 at 1 18 58 AM

**Comment by DanielSinclair** [comment]
Closing for now, but can revisit and reopen if the provider becomes more reliable

**Pull request closed without merging**
